### PR TITLE
feat: Record endpoints

### PR DESCRIPTION
**Pull request by duolok**
This pull request introduces endpoints for the host's PDF records. It's important to note that while these endpoints are defined in the code, their implementation is pending and will be addressed in future updates.

### CODE REVIEW

File: src/main/java/com/lunark/lunark/controller/RecordController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@RequestMapping("/api/records")
+public class RecordController {
+
+    @GetMapping("/download")
+    public ResponseEntity<String> downloadRecord(@RequestParam Long recordId) {
+       return new ResponseEntity<>("Dummy string", HttpStatus.OK);
+    }
+
+    @GetMapping("/send")
+    public ResponseEntity<String> sendRecordData(@RequestParam Long propertyId, @RequestParam String startDate, @RequestParam String endDate) {
```
Comment (MomirMilutinovic) [inline src/main/java/com/lunark/lunark/controller/RecordController.java:17]: Wouldn't it be more realistic if the data were returned in JSON?

File: src/main/java/com/lunark/lunark/controller/RecordController.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/records")
+public class RecordController {
+
+    @GetMapping("/download")
+    public ResponseEntity<String> downloadRecord(@RequestParam Long recordId) {
```
Comment (MomirMilutinovic) [inline src/main/java/com/lunark/lunark/controller/RecordController.java:12]: Shouldn't this method take the same parameters as the one for getting a record?

File: src/main/java/com/lunark/lunark/controller/RecordController.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.lunark.lunark.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/records")
+public class RecordController {
```
Comment (MomirMilutinovic) [inline src/main/java/com/lunark/lunark/controller/RecordController.java:9]: Would PropertyReportController be a better name?

**Review by MomirMilutinovic — CHANGES_REQUESTED** [review]

File: src/main/java/com/lunark/lunark/controller/RecordController.java (hunk 2 above)
Comment (duolok) [reply src/main/java/com/lunark/lunark/controller/RecordController.java:12]: I'm not sure how this will work.

Comment (MomirMilutinovic) [reply src/main/java/com/lunark/lunark/controller/RecordController.java]: The method signature would be public ResponseEntity<String> downloadRecord    public ResponseEntity<String> downloadRecord(@RequestParam Long recordId)

File: src/main/java/com/lunark/lunark/controller/PropertyReportController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+@RequestMapping("/api/property-report")
+public class PropertyReportController {
+
+    @GetMapping(value = "/download", produces =  "application/json")
+    public ResponseEntity<String> downloadRecord(@RequestParam Long propertyId, @RequestParam String startDate, @RequestParam String endDate) {
+        // TODO: Implementation
+       return ResponseEntity.ok("Dummy string.");
+    }
+
+    @GetMapping("/")
```
Comment (MomirMilutinovic) [inline src/main/java/com/lunark/lunark/controller/PropertyReportController.java:18]: This adds a trailing slash to the mappping /api/property-report/

File: src/main/java/com/lunark/lunark/controller/PropertyReportController.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    public ResponseEntity<String> sendRecordData(@RequestParam Long propertyId, @RequestParam String startDate, @RequestParam String endDate) {
```
Comment (MomirMilutinovic) [inline src/main/java/com/lunark/lunark/controller/PropertyReportController.java:19]: Shouldn't this return some sort of DTO?

**Review by MomirMilutinovic — CHANGES_REQUESTED** [review]

**Review by MomirMilutinovic — APPROVED** [review]

**Pull request merged**